posthog/tasks/alerts/checks.py

```diff
@@ -33,75 +39,99 @@ def check_all_alerts() -> None:
     group(groups).apply_async()
 
 
+@transaction.atomic
 def check_alert(alert_id: int) -> None:
-    alert = Alert.objects.get(pk=alert_id)
+    try:
+        alert = AlertConfiguration.objects.select_for_update().get(id=alert_id, enabled=True)
```

Comment:
I was referring to the inability to edit an alert in the UI when the task is in progress, e.g. querying ClickHouse. It might take minutes in the case of funnels. It's a minor problem though. I think this solution is reasonable.

Reply:
Ah I see. I didn't connect that in my head. Yeah that's a valid point. I'll try to lock the row after we have calculated.